### PR TITLE
chai-style: add types/tests

### PR DESCRIPTION
**Pull request by AviVahl**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/chai-style/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// Type definitions for chai-style 1.0
+// Project: https://github.com/darlanmendonca/chai-style
+// Definitions by: Avi Vahl <https://github.com/AviVahl>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.0
```
Comment (armanio123) [inline types/chai-style/index.d.ts]: I think you can remove this line as there's nothing using version 3.0

**Review by armanio123 — CHANGES_REQUESTED** [review]

File: types/chai-style/index.d.ts (hunk 1 above)
Comment (AviVahl) [reply types/chai-style/index.d.ts]: `@types/chai` themselves requires ts 3.0+:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/chai/index.d.ts

so I guessed any assertion plugin (that imports chai) also has that requirement. Is that not correct?

**Comment by AviVahl** [comment]
Removing the min typescript version fails the tests (locally):
```
Testing 1 changed packages: chai-style
Testing 0 dependent packages: 
Error: chai-style depends on chai but has a lower required TypeScript version.
    at checkTypeScriptVersions (/home/avi/projects/DefinitelyTyped/node_modules/types-publisher/src/check-parse-results.ts:51:23)
    at Object.checkParseResults [as default] (/home/avi/projects/DefinitelyTyped/node_modules/types-publisher/src/check-parse-results.ts:19:5)
    at main (/home/avi/projects/DefinitelyTyped/node_modules/types-publisher/src/tester/test.ts:26:9)
npm ERR! Test failed.  See above for more details.
```
@armanio123 any chance you could re-review? :)

File: types/chai-style/index.d.ts (hunk 1 above)
Comment (armanio123) [reply types/chai-style/index.d.ts]: Aha, makes sense. Thanks for checking.

**Review by armanio123 — APPROVED** [review]

**Pull request merged**